templates/examples/extensions/py_lambda_hooks/CanvasHook/CanvasHook.py

```diff
@@ -0,0 +1,405 @@
+            elif query == 'CourseEnrollments':
+                # Retrieve the course options for this student.
+                result = query_enrollments_for_student(canvas, student_user_name, userinput)
+                return_courses = result['CourseNames']
+                if return_courses:
+                    choicelist = []
+                    for i in return_courses:
+                        choicelist.append({'text':i,'value':"more information about my {} course".format(i)})
+                    genericAttachments = {'version': '1','contentType': 'application/vnd.amazonaws.card.generic','genericAttachments':[{"title":"response buttons","buttons":choicelist}]}
+                    event['res']['session']['appContext']['responseCard'] = genericAttachments
+                    event['res']['session']['appContext']['altMessages']['markdown'] = "Please select one of the options below:"
+                else:
+                    event['res']['session']['appContext']['altMessages']['markdown'] = "You are not currently enrolled in any courses"
+            elif query == 'SyllabusForCourse':
+                # Retrieve the course syllabus for this student.
+                result = query_syllabus_for_student(canvas, student_user_name, userinput)
+                event['res']['session']['appContext']['altMessages']['markdown'] = result['CourseSyllabus']
+            elif query == 'ChoicesForStudent':
+                # Retrieve the course options for this student.
+                result = query_choices_for_student(canvas, student_user_name, userinput)
+                returned_course = result['Choice']
+                genericattachment = ['assignments','syllabus','grades']
+                choicelist = []
+                for i in genericattachment:
+                    choicelist.append({'text':'{} {}'.format(returned_course,i),'value':'tell me about my {} {}'.format(returned_course,i)})
+                genericAttachments = {'version': '1','contentType': 'application/vnd.amazonaws.card.generic','genericAttachments':[{"title":"response buttons","buttons":choicelist}]}
+                event['res']['session']['appContext']['responseCard'] = genericAttachments
+            elif query == 'AnnouncementsForStudent':
+                # Retrieve the announcements for this student.
+                result = query_announcements_for_student(canvas, student_user_name, userinput)
+                if result['Announcements']:
+                    return_message = 'Here are your announcements: {}'.format(result['Announcements'])
+                    event['res']['session']['appContext']['altMessages']['markdown'] = return_message
+                else:
+                    event['res']['session']['appContext']['altMessages']['markdown'] = "You don't have any announcements at this moment."
+            elif query == 'GradesForStudent':
```

Comment:
Is there an `else` needed here if the query does not match any of the cases above. Or is validate_input guarding against that always?